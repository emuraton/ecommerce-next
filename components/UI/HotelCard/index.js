// @flow
import React from 'react';

import Card from './styles';

type Item = {
  image: string,
  name: string,
  highlights: Array<{ highlight: string }>,
  price: string,
  previousPrice: string,
};

type Props = {
  item: Item,
  onClick: (event: SyntheticMouseEvent<any>) => {},
  className?: string,
};

const HotelCard = ({ item, onClick, className }: Props) => {
  if (!item) return null;
  const {
    image, name, highlights, price, previousPrice,
  } = item;
  return (
    <Card.Container className={className}>
      <Card.ImageContainer>
        <Card.Image src={`/static/${image}`} />
      </Card.ImageContainer>
      <Card.FlexColumn>
        <Card.H2>{name}</Card.H2>
        <Card.List>
          {highlights.map((highlight, index) => (
            <Card.ListItem key={index}>{highlight}</Card.ListItem>
          ))}
        </Card.List>
        <div>
          <Card.Price>{price}</Card.Price>
          <Card.PerNight>per night</Card.PerNight>
        </div>
        <div>
          <Card.Price disabled>{previousPrice}</Card.Price>
          <Card.PerNight disabled>per night</Card.PerNight>
        </div>
        <Card.Button type="button" onClick={onClick}>
          Book Now
        </Card.Button>
      </Card.FlexColumn>
    </Card.Container>
  );
};

export default HotelCard;
