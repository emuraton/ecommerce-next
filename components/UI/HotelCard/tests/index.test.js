import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import HotelCard from '../';

describe('<HotelCard />', () => {
  const initialProps = {
    onClick: Function.prototype,
    item: {
      id: 44,
      image: 'shared-room.jpg',
      name: 'Shared room',
      highlights: [
        'Bunk bed in dormitory',
        'Private bathroom',
        'Flat screen TV',
      ],
      price: '$18.89',
      previousPrice: '$22',
    },
  };
  describe('@renders', () => {
    it('default render', () => {
      const wrapper = shallow(<HotelCard {...initialProps} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('@events', () => {
    // TODO refactor test once onClick actually do something
    it('onClick', () => {
      const mockedOnClick = jest.fn();
      const props = {
        ...initialProps,
        onClick: mockedOnClick,
      };
      const wrapper = shallow(<HotelCard {...props} />);
      const button = wrapper.find('Button');
      expect(mockedOnClick).toHaveBeenCalledTimes(0);
      button.simulate('click');
      expect(mockedOnClick).toHaveBeenCalledTimes(1);
    });
  });
});
