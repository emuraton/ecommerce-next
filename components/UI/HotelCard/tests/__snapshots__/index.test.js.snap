// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HotelCard /> @renders default render 1`] = `
<styles__Container
  className={null}
>
  <styles__ImageContainer>
    <styles__Image
      src="/static/shared-room.jpg"
    />
  </styles__ImageContainer>
  <styles__FlexColumn>
    <styles__H2>
      Shared room
    </styles__H2>
    <styles__List>
      <styles__ListItem
        key="0"
      >
        Bunk bed in dormitory
      </styles__ListItem>
      <styles__ListItem
        key="1"
      >
        Private bathroom
      </styles__ListItem>
      <styles__ListItem
        key="2"
      >
        Flat screen TV
      </styles__ListItem>
    </styles__List>
    <div>
      <styles__Price>
        $18.89
      </styles__Price>
      <styles__PerNight>
        per night
      </styles__PerNight>
    </div>
    <div>
      <styles__Price
        disabled={true}
      >
        $22
      </styles__Price>
      <styles__PerNight
        disabled={true}
      >
        per night
      </styles__PerNight>
    </div>
    <Button
      onClick={[Function]}
      type="button"
    >
      Book Now
    </Button>
  </styles__FlexColumn>
</styles__Container>
`;
